### PR TITLE
Upgrade to .NET 7, enable WASM optimizations

### PR DESCRIPTION
**Pull request by Arlodotexe**
Companion PR to https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/117, which should be closed before merging this.

As noted:
> Any codebase consuming the tooling repo will need to be adjusted to account for the new nullable annotations and analyzers added between netstandard2.0 and net7.0.

This PR pulls in the .NET 7 upgrade and fixes the new warnings and errors that appeared in our codebase as a result.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -433,5 +432,9 @@ csharp_style_prefer_extended_property_pattern = true:suggestion
 # Require file header
 dotnet_diagnostic.IDE0073.severity = warning
 
-# Uno platform exposes IDisposable on Storyboard publicly when it should be internal. Ignore this.
-dotnet_code_quality.CA1001.excluded_type_names_with_derived_types = T:Windows.UI.Xaml.Media.Animation.Storyboard
+# Member owns disposable field
+dotnet_diagnostic.CA2213.severity = none
+dotnet_diagnostic.CA1001.severity = none
+
+# Labs only: XML comment has param tag but no paramater by that name.
```
Comment (michael-hawker) [inline .editorconfig:439]: Isn't this a bug then on the comment that can be fixed?

File: .editorconfig [hunk 2]
```diff
@@ -433,5 +432,9 @@ csharp_style_prefer_extended_property_pattern = true:suggestion
 # Require file header
 dotnet_diagnostic.IDE0073.severity = warning
 
-# Uno platform exposes IDisposable on Storyboard publicly when it should be internal. Ignore this.
-dotnet_code_quality.CA1001.excluded_type_names_with_derived_types = T:Windows.UI.Xaml.Media.Animation.Storyboard
+# Member owns disposable field
+dotnet_diagnostic.CA2213.severity = none
+dotnet_diagnostic.CA1001.severity = none
```
Comment (michael-hawker) [inline .editorconfig:437]: Should have a comment here labeling what this is, it's helpful. 🙂

File: .github/workflows/build.yml
Comment (michael-hawker) [inline .github/workflows/build.yml:1]: Should diff and pull-over other updates to the build.yml, like pushing to the PR feed and such.

File: components/DataTable/samples/DataTable.Samples.csproj
Comment (michael-hawker) [inline components/DataTable/samples/DataTable.Samples.csproj:1]: Why is this deleted?

File: components/DataTable/src/MultiTarget.props
Comment (michael-hawker) [inline components/DataTable/src/MultiTarget.props:1]: Why is this deleted? Or did Uno add support for the ItemsPresenter Header/Footer?

File: components/StackedNotificationsBehavior/src/Notification.cs
Comment (michael-hawker) [inline components/StackedNotificationsBehavior/src/Notification.cs:1]: We can delete this whole component as it was migrated to the main repo. Is Segmented still here too, as we migrated that one too, right? Feel free to do a quick compare and pass and clean-up anything no longer needed here.

**Review by michael-hawker — CHANGES_REQUESTED** [review]
Looks good, few things to tweak. Still a low-priority for this though with the focus on the main release.

We should link a tracking issue for the investigation on the fixed .NET 7 version from the tooling repo.

File: components/DataTable/samples/DataTable.Samples.csproj
Comment (Arlodotexe) [reply components/DataTable/samples/DataTable.Samples.csproj:1]: Good question, that shouldn't have happened. I'll restore it.

File: components/StackedNotificationsBehavior/src/Notification.cs
Comment (Arlodotexe) [reply components/StackedNotificationsBehavior/src/Notification.cs:1]: Good catch, filed a ticket here -> https://github.com/CommunityToolkit/Labs-Windows/issues/489

File: components/DataTable/src/MultiTarget.props
Comment (Arlodotexe) [reply components/DataTable/src/MultiTarget.props:1]: This shouldn't have happened. I'll need to restore these.

File: .editorconfig [hunk 3]
```diff
@@ -433,5 +432,6 @@ csharp_style_prefer_extended_property_pattern = true:suggestion
 # Require file header
 dotnet_diagnostic.IDE0073.severity = warning
 
-# Uno platform exposes IDisposable on Storyboard publicly when it should be internal. Ignore this.
-dotnet_code_quality.CA1001.excluded_type_names_with_derived_types = T:Windows.UI.Xaml.Media.Animation.Storyboard
+# Member owns disposable field
+dotnet_diagnostic.CA2213.severity = none
```
Comment (Arlodotexe) [inline .editorconfig:436]:
```suggestion
# Todo: Clean up suppression of CA1063 and implement IDisposable correctly
# https://github.com/CommunityToolkit/Labs-Windows/issues/443
dotnet_diagnostic.CA2213.severity = none
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -131,16 +167,16 @@ jobs:
...
+        run: ./PackEachExperiment.ps1 -extraBuildProperties "${{ env.VERSION_PROPERTY }}"
+        
+      # Push Pull Request Packages to our DevOps Artifacts Feed (see nuget.config)
+      - name: Push Pull Request Packages (if not fork)
+        if: ${{ env.IS_PR == 'true' && github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]' }}
+        run: |
+          dotnet nuget add source https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-PullRequests/nuget/v3/index.json `
+            --name PullRequests `
+            --username dummy --password ${{ secrets.DEVOPS_PACKAGE_PUSH_TOKEN }}
+          dotnet nuget push "**/*.nupkg" --api-key dummy --source PullRequests --skip-duplicate
```
Comment (michael-hawker) [inline .github/workflows/build.yml:179]: If it's not a PR, then we still need to push to the LabsFeed, as there's no other steps like in the main build that will push to the feed, right?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -116,13 +131,34 @@ jobs:
...
+        run: |
+          echo "VERSION_DATE=$(git log -1 --format=%cd --date=format:%y%m%d)" >> $env:GITHUB_ENV
+      # Semver regex: https://regex101.com/r/Ly7O1x/3/
+      - name: Format Date/Time of Release Package Version
+        if: ${{ env.IS_RELEASE == 'true' }}
+        run: |
+          $ref = "${{ github.ref }}"
+          $ref -match "^refs/heads/rel/(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:-(?<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+          echo "VERSION_DATE=$($matches['patch'])" >> $env:GITHUB_ENV
+          echo "VERSION_PROPERTY=$($matches['prerelease'])" >> $env:GITHUB_ENV
```
Comment (michael-hawker) [inline .github/workflows/build.yml:149]: Since this will always be false, do we just want to exclude it?

**Comment by michael-hawker** [comment]
@Arlodotexe can you see if any of the open PRs need updating from this? Thanks!

**Comment by michael-hawker** [comment]
Wasn't sure why GitHub wasn't reporting anything, had to dig-in to the Actions page for the last build and see the 'Startup Failure' seems like one of the actions isn't on the allowlist, going to fix that.

**Comment by michael-hawker** [comment]
Going to close and re-open to kick off CI again

**Comment by michael-hawker** [comment]
@Arlodotexe CI is running again, but something up with the call to the pack script:

```
   2 |  ./PackEachExperiment.ps1 -extraBuildProperties "pull-485.1577"
     |                           ~~~~~~~~~~~~~~~~~~~~~
     | A parameter cannot be found that matches parameter name 'extraBuildProperties'.
```

Maybe the sub-module isn't updated or something? Should hopefully be an easy fix of some sort either way.

**Comment by Jay-o-Way** [comment]
Hey guys, check out the MS video: "Upgrade your .NET projects with Visual Studio"
https://build.microsoft.com/en-US/sessions/c5bf6653-f0de-4091-a139-caf7904300c6

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -132,12 +154,16 @@ jobs:
       # Build All Packages
       - name: pack experiments
         working-directory: ./tooling/Scripts/
-        run: ./PackEachExperiment.ps1 all
-
-      # Push Packages to our DevOps Artifacts Feed
-      - name: Add source
-        if: ${{ github.ref == 'refs/heads/main' }}
-        run: dotnet nuget update source LabsFeed --username dummy --password ${{ secrets.DEVOPS_PACKAGE_PUSH_TOKEN }}
```
Comment (michael-hawker) [inline .github/workflows/build.yml:140]: @Arlodotexe this is still needed in between the added PR step and the 'Push packages' still. This is why `main` is failing, as the source isn't updated to include the token.

**Pull request merged**